### PR TITLE
Livesite Schedule

### PR DESCRIPTION
**Pull request by ianmah**
Mostly copied implementation from /[id]/events.js

image
image

### CODE REVIEW

File: pages/Livesite/schedule.js [hunk 1]
```diff
@@ -1,23 +1,493 @@
...
+  }, [alertMsg]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, [activeHackathon]);
+
+  const handleNew = async () => {
+    newEvent.lastModifiedBy = user;
+    newEvent.startTime = new Date(newEvent.startTime).toISOString();
+    newEvent.endTime = new Date(newEvent.endTime).toISOString();
```
Comment (panjenny0) [inline pages/Livesite/schedule.js:82]:
```suggestion
    newEvent.startTime = new Date(newEvent.startTime)
    newEvent.endTime = new Date(newEvent.endTime)
```

File: pages/Livesite/schedule.js [hunk 2]
```diff
@@ -1,23 +1,493 @@
...
+    const eventID = await addLivesiteEvent(activeHackathon, { ...newEvent });
+    newEvent.lastModified = formatDate(getTimestamp().seconds);
+    setEvents({
+      ...events,
+      [eventID]: {
+        ...newEvent,
```
Comment (panjenny0) [inline pages/Livesite/schedule.js:88]: I think `startTime` and `endTime` should be formatted here for consistency with the other pages. You probably saw this in events - I basically did `date: formatDate(eventEditing.date, true),`

I think the same should be applied to `handleUpdate`

File: utility/firebase.js [hunk 3]
```diff
@@ -455,3 +455,72 @@ export const subscribeToLivesiteData = (callback) => {
...
+        ...data,
+        eventID,
+        key: data.key || eventID,
+        name: data.name || 'Empty event field',
+        description: data.description || 'Empty text description for event',
+        lastModified: data.lastModified
+          ? formatDate(data.lastModified.seconds)
+          : formatDate(getTimestamp().seconds),
+        lastModifiedBy: data.lastModifiedBy || 'Unknown user',
+      }
```
Comment (panjenny0) [inline utility/firebase.js:471]: I think `startTime` and `endTime` should be formatted here - I recently refactored the `formatDate` helper so it should work now

**Review by panjenny0 — CHANGES_REQUESTED** [review]
Just added some comments regarding the formatting of the timestamps since I think it'd be good to keep them consistent with the other pages. My refactoring of `formatDate` earlier should've fixed the issues with it 😛

**Comment by ianmah** [comment]
Totally agree with you that we should have consistent dates @panjenny0 however the livesite schedule has already been implemented using ISO date strings and I do not want someone to have to go and refactor the livesite

**Comment by panjenny0** [comment]
> Totally agree with you that we should have consistent dates @panjenny0 however the livesite schedule has already been implemented using ISO date strings and I do not want someone to have to go and refactor the livesite

Oh I see, ok! `formatDate` is only changes the timestamp on the frontend though - not sure how this affects livesite?

**Review by panjenny0 — APPROVED** [review]

**Comment by jackyzha0** [comment]
Can we change type to be a dropdown?

**Comment by ianmah** [comment]
image

**Comment by ianmah** [comment]
image

**Review by jackyzha0 — APPROVED** [review]
looks good chief 🧼 🧞

**Pull request merged**